pkg/maps/lbmap/source_range.go

```diff
@@ -55,7 +55,7 @@ type SourceRangeKey4 struct {
 func (k *SourceRangeKey4) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
 func (k *SourceRangeKey4) NewValue() bpf.MapValue    { return &SourceRangeValue{} }
 func (k *SourceRangeKey4) String() string            { return fmt.Sprintf("%s", k.Address) }
-func (k *SourceRangeKey4) ToNetwork() *SourceRangeKey4 {
+func (k *SourceRangeKey4) ToNetwork() SourceRangeKey {
```

Comment:
another pattern I came to know recently is something like below, this will enforce that a struct statisfied particular interface.

```
var _ SourceRangeKey = (*SourceRangeKey4)(nil)
var _ SourceRangeKey = (*SourceRangeKey6)(nil)
```